test/browser/third-party/axios-upload.browser.test.ts

```diff
@@ -0,0 +1,35 @@
+import type { AxiosResponse } from 'axios'
+import { test, expect } from '../playwright.extend'
+
+declare namespace window {
+  export let upload: () => Promise<
+    AxiosResponse<{ message: string; content: string }>
+  >
+  export let progressEvents: Array<ProgressEvent>
+}
+
+test('responds with a mocked response to an upload request', async ({
+  loadExample,
+  page,
+}) => {
+  await loadExample(require.resolve('./axios-upload.runtime.js'))
+
+  const uploadResult = await page.evaluate(() => {
+    return window.upload().then((response) => response.data)
+  })
+
+  expect(uploadResult).toEqual({
+    message: 'Successfully uploaded "doc.txt"!',
+    content: 'Helloworld',
+  })
+
+  const progressEvents = await page.evaluate(() => {
+    return window.progressEvents
+  })
+
+  expect(progressEvents.length).toBeGreaterThan(0)
+  expect(progressEvents[0]).toMatchObject({
```

Comment:
How many progress events Axios fires is not something we should assert. As long as at least one is fired, we can conclude the upload behaved correctly.